app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,208 @@
+        assertTrue(onView(withSubstring(formatted_date)).isPresent())
+        assertTrue(onView(withSubstring(date_selected)).isPresent())
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+    }
+
+
+    @Test
+    fun testEthiopianDateWidgets(){
+
+        InstrumentationUtility.openModule(1)
+        onView(withSubstring("This form will test the Ethiopian date widget.")).isPresent()
+        InstrumentationUtility.nextPage()
+        InstrumentationUtility.nextPage()
+        for (i in 1..12){
+            val month_text = InstrumentationUtility.getText(onView(withId(R.id.monthtxt)))
+            if (month_text != null) {
+                assertTrue(listOfEthiopianMonths.contains(month_text))
+            }
+            onView(withId(R.id.monthupbtn)).perform(ViewActions.click())
+        }
+
+        onView(withId(R.id.yeardownbtn)).perform(ViewActions.click())
+        val date_selected = setDateToUniversalCalender(-10,0,0)
+        val gregorian_date = InstrumentationUtility.getText(onView(withId(R.id.dateGregorian)))
+        val formatted_date = formatGregorianDate(gregorian_date.drop(1).dropLast(1))
+
+        InstrumentationUtility.nextPage()
+        Log.i("Ethiopian date:", date_selected)
+        Log.i("Gregorian date:", formatted_date)
+        assertTrue(onView(withSubstring(formatted_date)).isPresent())
+        assertTrue(onView(withSubstring(date_selected)).isPresent())
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+    }
+
+    @Test
+    fun testStandardWidget(){
+
+        InstrumentationUtility.openModule(4)
+        InstrumentationUtility.rotateLeft()
+        val dateList = setDateTo(-10)
+//        Thread.sleep(1000)
+        InstrumentationUtility.rotatePortrait()
+        InstrumentationUtility.nextPage()
+
+        for (formats in dateList){
+            assertTrue(onView(allOf(withSubstring(formats))).isPresent())
+        }
+
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+    }
+
+    /**
+     * function to set date in the date picker and return different format of the selected date
+     */
+    fun setDateTo(days: Int): Array<String> {
+        val calendar = Calendar.getInstance()
+        calendar.add(Calendar.DATE, +days)
+
+        val year: Int = calendar.get(Calendar.YEAR)
+        val month: Int = calendar.get(Calendar.MONTH) + 1
+        val day: Int = calendar.get(Calendar.DAY_OF_MONTH)
+        onView(
+            ViewMatchers.withClassName(
+                Matchers.equalTo(
+                    DatePicker::class.java.name
+                )
+            )
+        ).perform(PickerActions.setDate(year, month, day))
+
+        val long_format = SimpleDateFormat("EE, MMM dd, yyyy").format(calendar.time)
```

Comment:
it seems like this function is doing 2 things , one setting date on date picket and second returning formats. It would be nice to split these into 2 separate functions like - 

`fun setDatePicker(days: Int) ` : this can return `calendar.time` 
`fun getDateInDifferentFormats(time : Long) : Array<String>` , you should call this with result of `setDatePicker`

Reply:
Done